packages/react-liveness/src/components/FaceLivenessDetector/service/utils/TelemetryReporter/TelemetryReporter.ts

```diff
@@ -0,0 +1,69 @@
+// type TelemetryDataKey = `${Lowercase<string>}-${Lowercase<string>}`;
+// type TelemetryData = Record<TelemetryDataKey, string>;
+import {
+  StartFaceLivenessSessionCommandInput,
+  StartFaceLivenessSessionCommandOutput,
+} from '@aws-sdk/client-rekognitionstreaming';
+import {
+  BuildHandler,
+  BuildHandlerArguments,
+  BuildHandlerOutput,
+} from '@smithy/types';
+
+export const ATTEMPT_KEY = 'AmplifyLivenessAttempt';
+const DEFAULT_ATTEMPT_COUNT_TIMEOUT = 300000; // 5 minutes / 300000 ms
+
+type AttemptCountData = { timestamp: number; count: number };
+
+function getAttemptCountData(): AttemptCountData {
+  return JSON.parse(localStorage.getItem(ATTEMPT_KEY) ?? '{}') as {
+    timestamp: number;
+    count: number;
+  };
+}
+
+function setAttemptCountData(data: AttemptCountData) {
```

Comment:
Opted to use local storage so that the attempt count could be tracked between attempts. In cases where the component gets unmounted and then re mounted the liveness state machine would be reset and we have no way of tracking attempts. This is a common use case in even in our default example app as we show a loading spinner in place of the component while we grab the session id

Reply:
Let's discuss oflline